doc/apt-repo.md

```diff
@@ -21,7 +21,7 @@ If you like to live dangerously, these are the bleeding edge builds that are gen
 Things may break! But if they do and you report it to us, you'll be a hero to CKAN users everywhere, whether they know it or not.
 
 ```
-sudo curl -sS -o /etc/apt/trusted.gpg.d/ksp-ckan.gpg https://raw.githubusercontent.com/KSP-CKAN/CKAN/master/debian/ksp-ckan.gpg
-sudo apt-add-repository -u -y 'deb https://ksp-ckan.s3-us-west-2.amazonaws.com/deb nightly main'
+sudo curl -sS -o /usr/share/keyrings/ckan-archive-keyring.gpg https://raw.githubusercontent.com/KSP-CKAN/CKAN/master/debian/ksp-ckan.gpg
+sudo apt-add-repository -u -y 'deb [arch=amd64 signed-by=/usr/share/keyrings/ckan-archive-keyring.gpg] https://ksp-ckan.s3-us-west-2.amazonaws.com/deb nightly main'
```

Comment:
Thanks so much for the update!

This command doesn't work on my Ubuntu 20.04 VM:

```
$ sudo apt-add-repository -u -y 'deb [arch=amd64 signed-by=/usr/share/keyrings/ckan-archive-keyring.gpg] https://ksp-ckan.s3-us-west-2.amazonaws.com/deb nightly main'
[sudo] password for user: 
Error: 'deb [arch=amd64 signed-by=/usr/share/keyrings/ckan-archive-keyring.gpg] https://ksp-ckan.s3-us-west-2.amazonaws.com/deb nightly main' invalid
```

Any idea what's wrong?

Reply:
Hmm, it looks like there's a bug in `apt-add-repository` which is preventing it from parsing that deb line correctly. I personally wrote/maintain the parser used in Pop_OS, and adding it with my tool it does work correctly (Apt tells me it's verified and updates/installs work correctly) with the repo configured as in the PR.

A possible alternative is adding the repo using cat and tee, like so:
```
cat "deb [arch=amd64 signed-by=/usr/share/keyrings/ckan-archive-keyring.gpg] https://ksp-ckan.s3-us-west-2.amazonaws.com/deb nightly main" | sudo tee /etc/apt/sources.list.d/ckan.list
```
This would, however, require running `sudo apt update` afterwards to fetch the updated lists. 

Another option would be to put together a `.list` (or a DEB822-formatted `.sources`) file in the repo along side the key, which the user can install using curl similar to installing the key. This would also require `sudo apt update` following to fetch the lists, but it would be a bit cleaner looking I think. Downside would be that it's a bit less transparent to the user what exactly is being added, but it's not major. 

In either case, the key should change and the "signed-by" option added to point at the key. trusted.gpg.d is deprecated and will not be supported for much longer in the future.

Reply:
OK, that makes sense. Could you please pick an option and update this branch? We can't merge a PR that would have most of the affected users reporting errors in a process that worked previously.

Reply:
The `apt-add-repository` issue seems to be tracked here:

https://bugs.launchpad.net/ubuntu/+source/software-properties/+bug/1862764

Reply:
Pushed some changes that resolve this (`echo ... | sudo tee`). I tried them in my test VM and they worked.